geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -1,35 +1,17 @@
 import { Injectable, Logger } from '@nestjs/common';
 
-import { createReadStream, rmdirSync, unlinkSync } from 'fs';
-import { Extract } from 'unzipper';
+import { FileService } from '../../utils/file.utils';
+
 const mapshaper = require('mapshaper');
 
 @Injectable()
-export class ShapeFileService {
-  constructor(private readonly logger: Logger) {
-    this.logger.setContext(ShapeFileService.name);
-  }
-
-  private unzipShapefile(fileInfo: Express.Multer.File): Promise<string> {
-    return new Promise((resolve, reject) => {
-      createReadStream(fileInfo.path)
-        .pipe(
-          Extract({
-            path: `${fileInfo.destination}/${fileInfo.filename.replace(
-              '.zip',
-              '',
-            )}`,
-          }),
-        )
-        .on('close', () =>
-          resolve(`${fileInfo.filename} extracted successfully`),
-        )
-        .on('error', (error: Error) =>
-          reject(
-            new Error(`${fileInfo.filename} could not be extracted: ` + error),
-          ),
-        );
-    });
+export class ShapefileService {
+  constructor(
+    private readonly logger: Logger,
+    private fileService: FileService,
+  ) {
+    this.logger.setContext(ShapefileService.name);
+    this.fileService = new FileService();
```

Comment:
Why are we bypassing NestJS's DI ?

Reply:
@kgajowy 1d15b56c1ef8c605a970984b80f2616a35a65f64

Reply:
@alexeh 
```
 @Inject(FileService)    private fileService: FileService
```

Should be enough to `private readonly fileService: FileService` as long as you import the FileModule

Reply:
@kgajowy 1105fe35e9b4ce2eeb8265172f46496f055d8978